vsl-sdk/src/lib.rs

```diff
@@ -55,14 +86,66 @@ impl Amount {
             None
         }
     }
+
     pub const fn checked_add(self, rhs: Amount) -> Option<Amount> {
         if let Some(v) = self.0.checked_add(rhs.0) {
             Some(Amount(v))
         } else {
             None
         }
     }
+
+    pub fn to_hex_str(&self) -> String {
+        format!("{:#x}", self)
+    }
+
+    pub fn from_hex_str(s: &str) -> Result<Self, ParseAmountError> {
+        if !s.starts_with("0x") {
+            return Err(ParseAmountError::NotHex);
+        }
+        let s = &s[2..];
+        if s == "0" {
+            return Ok(Self::ZERO);
+        }
+        if s.starts_with('0') {
+            return Err(ParseAmountError::LeadingZeros);
+        } else {
+            let subunits =
+                u128::from_str_radix(s, 16).map_err(ParseAmountError::ParseInt)?;
+            return Ok(Self::from_subunits(subunits));
+        }
+    }
+
+    pub fn to_str_with_decimals(&self, decimals: u8) -> String {
+        let one_token = 10u128.checked_pow(decimals as u32).expect("overflow");
+        let units = self.0 / one_token;
+        let subunits = self.0 % one_token;
+        format!("{}.{:0>width$}", units, subunits, width = decimals as usize)
+    }
+
+    pub fn from_str_with_decimals(s: &str, decimals: u8) -> Result<Self, ParseAmountError> {
+        let mut iter = s.split(".");
+        let Some(units) = iter.next() else {
+            return Err(ParseAmountError::NotDecimal)
+        };
+        let subunits = iter.next().unwrap_or("0");
+        if iter.next().is_some() {
+            return  Err(ParseAmountError::NotDecimal);
+        }
+        let decimals = decimals as usize;
+        let subunits = match subunits.len().cmp(&decimals) {
+            std::cmp::Ordering::Equal => subunits.to_string(),
+            std::cmp::Ordering::Less => format!("{:0<width$}", subunits, width = decimals),
+            std::cmp::Ordering::Greater => {
+                return  Err(ParseAmountError::TooManyDecimals);
+            }
+        };
+        let restored = format!("{}{}", units, subunits);
+        let amount = u128::from_str_radix(&restored, 10).map_err(ParseAmountError::ParseInt)?;
```

Comment:
Doesn't look nice to make one String from subunits then immediately format! it into a larger string, but should at least function correctly, and I don't think we expect decimal parsing anywhere performance matters.